src/app/recommended-communities/recommended-communities.component.ts

```diff
@@ -16,36 +16,70 @@ import { CommunityUserResponse } from '../communities/community-user-response';
 })
 export class RecommendedCommunitiesComponent implements OnInit {
 
-  communities$: Observable<CommunityResponse[]> = of([]);
+  communities: CommunityResponse[] = [];
```

Comment:
I think I got the reason why you turned `Observable` into a simple array. And this is a necessity to filter joined community out. But could we stay with `Observable` and filter it with https://www.learnrxjs.io/operators/filtering/filter.html ? After a user joined a community we can replace the original `Observable` with a filtered one. And when the component is refreshed the new list of recommended communities without a joined one will be provided by the server. What do you think about it?